ts/src/htx.ts

```diff
@@ -1939,6 +1919,44 @@ export default class htx extends Exchange {
         return result;
     }
 
+    trySetSymbolFromFutureMarkets (symbolOrMarketId: string) {
+        if (symbolOrMarketId in this.markets) {
+            return symbolOrMarketId;
+        }
+        // only on "future" market type (inverse & linear), market-id differs between "fetchMarkets" and "fetchTicker"
+        // so we have to create a mapping
+        // - market-id from fetchMarkts:    `BTC-USDT-240419` (linear future) or `BTC240412` (inverse future)
+        // - market-id from fetchTciker[s]: `BTC-USDT-CW`     (linear future) or `BTC_CW`    (inverse future)
+        if (!('futureMarketIdsForSymbols' in this.options)) {
+            this.options['futureMarketIdsForSymbols'] = {};
+        }
+        const futureMarketIdsForSymbols = this.safeDict (this.options, 'futureMarketIdsForSymbols', {});
+        if (symbolOrMarketId in futureMarketIdsForSymbols) {
+            return futureMarketIdsForSymbols[symbolOrMarketId];
+        }
+        const futureMarkets = this.filterBy (this.markets, 'future', true);
```

Comment:
This is a bit worrying, if you call fetchTickers and you need to parse hundreds of markets then you will call this filterBy hundreds of time (which is going to slow down parsing)

Reply:
if i correctly understand, 
I created a temporary dictionary (`this.options['futureMarketIdsForSymbols']`)  so in this method, that filtering happens only once per instance lifetime for any market-id, so after it is "detected" once, it's saved in `futureMarketIdsForSymbols` dict, and then in the top lines of this method, on next calls, it's quickly resolved. so, that "filter" happens only initially on first "fetchTickers" call

Reply:
but you might be right - i.e. as I see now, the "undetected" symbols might be skipped from that clause, where I save marketids into temp dict. so, unsaved marketIds might cause re-filtering on next calls too. I'll push a minor commit for that too

Reply:
I've pushed and now i can confirm that it's only called once on first call of `fetchTickers` and then the filter lines are never reached again.
about memory consumption, only this small dict is created: 
![2024-04-08 22 01 49](https://github.com/ccxt/ccxt/assets/7117978/171b5956-f6fb-41b0-9b3c-b9645d470b82)